### ISSUE TITLE
[ML] GetJobStatsAction does not match open jobs correctly

### ISSUE THREAD

**Issue by davidkyle** [elastic/elasticsearch#34745]
The endpoint accepts a comma separated list of job Ids, group Ids, wildcarded Ids or `_all` which is then [expanded](https://github.com/elastic/elasticsearch/blob/master/x-pack/plugin/ml/src/main/java/org/elasticsearch/xpack/ml/action/TransportGetJobsStatsAction.java#L72) into individual job Ids. But the [task match function](https://github.com/elastic/elasticsearch/blob/master/x-pack/plugin/core/src/main/java/org/elasticsearch/xpack/core/ml/action/GetJobsStatsAction.java#L97) works on the unexpanded job id in the request rather than each of the expanded Ids. `_all` is handled correctly.

There are test cases covering getting the job stats for multiple jobs but the tests pass because the code assumes the job is closed then goes and gets the job stats for the job as if it was closed.

**Comment by MichaHoffmann** [comment]
As a first timer, i'd like to try this one!

**Comment by dimitris-athanasiou** [comment]
Go for it @MichaHoffmann!

**Comment by klobin** [comment]
I am first timer to Open source technologies, can I git it a shot ??

**Comment by droberts195** [comment]
@MichaHoffmann are you working on this issue?  If so I'll assign it to you to make that clearer.

@klobin if @MichaHoffmann is no longer planning to work on this then you're welcome to take it.  But please don't start work until that's clear because I'd hate one of you to waste effort on duplication.

**Comment by ricardojaferreira** [comment]
If no one is working on this I would like to give it a try.

**Comment by klobin** [comment]
@MichaHoffmann  Do let me know, if you still work on it as I got the solution and refactored the code.

I will PR or I will look up for another bug fix, to merge into develop or master.

**Comment by up201608320** [comment]
@davidkyle When you said that:'There are test cases covering getting the job stats for multiple jobs but the tests pass because the code assumes the job is closed then goes and gets the job stats for the job as if it was closed.' Can you specify or redirect me which tests were you talking about?
